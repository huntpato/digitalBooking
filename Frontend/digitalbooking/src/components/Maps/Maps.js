import React, { useState, useEffect } from "react";
import {
  GoogleMap,
  LoadScript,
  Marker,
  InfoWindow,
} from "@react-google-maps/api";

export const Maps = (props) => {
  const { lat, long, city, name, width, height, margin } = props;
  const [defaultCenter, setCenter] = useState({
    lat: 58.5558,
    lng: 60.5487,
  });
  const [visible, setVisible] = useState(false); 
 

  useEffect(() => {
    setCenter({ lat: parseFloat(lat), lng: parseFloat(long) });
  }, [lat, long]);

  const mapStyles = {
    height: height,
    width: width ,
    position: "relative",
    margin: margin,
  };

  const handleClick = (e) =>{
    setVisible(true);
  }
   
  return (
    <LoadScript googleMapsApiKey="AIzaSyCCK3XtE2nkcDKC5P8jaUV6Q-A8-nfXjFI">
      <GoogleMap mapContainerStyle={mapStyles} zoom={14} center={defaultCenter} onClick={() => setVisible(false)}>
        <Marker
          key="marker_1"
          position={{
            lat: parseFloat(lat),

            lng: parseFloat(long),
          }}
          onClick= {handleClick}
        />
           {visible && <InfoWindow position={{
            lat: parseFloat(lat + 1),

            lng: parseFloat(long + 1),
          }}   onCloseClick={() => setVisible(false)}
          options={{
            pixelOffset: new window.google.maps.Size(
              0, -30
            )
          }}>
            <div>
              <h4>{name}</h4>
              <p>{city}</p>
            </div>
        </InfoWindow>}

      </GoogleMap>
    </LoadScript>
  );
};
