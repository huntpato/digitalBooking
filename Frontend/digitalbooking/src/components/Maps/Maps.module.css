.mapStyles{

    padding-left:"0px";
    
}