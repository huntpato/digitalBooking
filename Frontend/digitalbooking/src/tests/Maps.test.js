import React from 'react';
import { shallow } from 'enzyme';
import { Maps } from '../components/Maps/Maps'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <Maps />', ()=> {

    
    const wrapper= shallow(<Maps lat="53.02" long="-45.02" width= "94%" height="80vh"margin= "3%" />);

    test('Comprobar que Maps renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    
})
