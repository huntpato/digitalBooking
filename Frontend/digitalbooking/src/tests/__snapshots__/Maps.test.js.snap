// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Maps /> Comprobar que Maps renderiza correctamente. 1`] = `
<LoadScript
  googleMapsApiKey="AIzaSyCCK3XtE2nkcDKC5P8jaUV6Q-A8-nfXjFI"
  id="script-loader"
  version="weekly"
>
  <GoogleMap
    center={
      Object {
        "lat": 58.5558,
        "lng": 60.5487,
      }
    }
    mapContainerStyle={
      Object {
        "height": "80vh",
        "margin": "3%",
        "position": "relative",
        "width": "94%",
      }
    }
    onClick={[Function]}
    zoom={14}
  >
    <Marker
      key="marker_1"
      onClick={[Function]}
      position={
        Object {
          "lat": 53.02,
          "lng": -45.02,
        }
      }
    />
  </GoogleMap>
</LoadScript>
`;
